tests/test_ophys/test_tools_roiextractors.py

```diff
@@ -1054,6 +1055,61 @@ def test_add_fluorescence_traces_regular_timestamps(self):
             self.assertEqual(roi_response_series[series_name].starting_time, times[0])
             self.assertEqual(roi_response_series[series_name].timestamps, None)
 
+    def test_add_fluorescence_traces_regular_timestamps_with_metadata(self):
+        """Test adding traces with regular timestamps and also metadata-specified rate."""
+        times = np.arange(0, 5)
+        segmentation_extractor = generate_dummy_segmentation_extractor(
+            num_rois=2,
+            num_frames=5,
+            num_rows=self.num_rows,
+            num_columns=self.num_columns,
+        )
+        segmentation_extractor.set_times(times)
+
+        metadata = deepcopy(self.metadata)
+        metadata["Ophys"]["Fluorescence"]["roi_response_series"][0].update(rate=1.23)
+        metadata["Ophys"]["DfOverF"]["roi_response_series"][0].update(rate=1.23)
+
+        add_fluorescence_traces(
+            segmentation_extractor=segmentation_extractor,
+            nwbfile=self.nwbfile,
+            metadata=metadata,
+        )
+
+        ophys = get_module(self.nwbfile, "ophys")
+        roi_response_series = ophys.get(self.fluorescence_name).roi_response_series
+        for series_name in roi_response_series.keys():
+            self.assertEqual(roi_response_series[series_name].rate, 1.23)
+            self.assertEqual(roi_response_series[series_name].starting_time, 0)
+            self.assertEqual(roi_response_series[series_name].timestamps, None)
+
+    def test_add_fluorescence_traces_irregular_timestamps_with_metadata(self):
```

Comment:
OK, so if I got you right, the behavior is that if the user does pass the metadata (in this case rate) then this should overwrite any other behavior. That is, even if the timestamps are indeed irregular the file will be written as a regular one. 

I modified the code so it does this.

Reply:
Thanks, I'll check it out in practice